src/parser.cpp

```diff
@@ -1332,14 +1333,21 @@ const Statement *Parser::parseVarStatement(Scope *scope, int line)
         } else {
             v = new LocalVariable(name, vars.second, scope, false);
         }
-        scope->addName(name, v);
+        variables.insert(std::map<std::string, Variable *>::value_type(name, v));
+        
         varrefs.push_back(new VariableExpression(v));
```

Comment:
You're setting up `varrefs` here but only use it in the case where an assignment is happening. Suggest just creating a list of `Variable *`, as `variables` that you already have. Then, in the case where you need to create an `AssignmentStatement`, create `varrefs` at that point.